src/com/amazon/ion/benchmark/schema/ReparsedType.java

```diff
@@ -0,0 +1,115 @@
+package com.amazon.ion.benchmark.schema;
+
+import com.amazon.ion.IonStruct;
+import com.amazon.ion.IonType;
+import com.amazon.ion.IonValue;
+import com.amazon.ion.benchmark.schema.constraints.*;
+import com.amazon.ionschema.Type;
+
+import java.util.HashMap;
+import java.util.Map;
+
+// Parsing the type definition in ISL file into ReparsedType format which allows getting constraints information directly.
+public class ReparsedType {
+    public final Type type;
+    private static final String KEYWORD_TIMESTAMP_PRECISION = "timestamp_precision";
+    private static final String KEYWORD_TYPE = "type";
+    private static final String KEYWORD_CODE_POINT_LENGTH = "codepoint_length";
+    private static final String KEYWORD_REGEX = "regex";
+    private static final String KEYWORD_CONTAINER_LENGTH = "container_length";
+    private static final String KEYWORD_BYTE_LENGTH = "byte_length";
+    private static final String KEYWORD_SCALE = "scale";
+    private static final String KEYWORD_PRECISION = "precision";
+    private static final String KEYWORD_VALID_VALUES = "valid_values";
+    private static final String KEYWORD_NAME = "name";
+    // Using map to avoid processing the multiple repeat constraints situation.
+    Map<String, ReparsedConstraint> constraintMap;
+
+    /**
+     * Initializing the newly created ReparsedType object.
+     * @param type represents type definition of ISL file.
+     */
+    public ReparsedType(Type type) {
+        this.type = type;
+        constraintMap = new HashMap<>();
+        getIsl().forEach(this::handleField);
+    }
+
+    /**
+     * Get the name of type definition.
+     * @return the name of type definition.
+     */
+    public String getName() {
+        return type.getName();
+    }
+
+    /**
+     * Handling the fields which are not used for specifying generated data.
+     * @param field represents the field contained by the type definition.
+     */
+    private void handleField(IonValue field) {
+        switch (field.getFieldName()) {
+            case KEYWORD_NAME:
+            case KEYWORD_TYPE:
+                return;
+            default:
+                constraintMap.put(field.getFieldName(), toConstraint(field));
+        }
+    }
+
+    /**
+     * Redefining the getIsl method to convert type definition to IonStruct format.
+     * @return an IonStruct which contains constraints in type definition.
+     */
+    public IonStruct getIsl() {
+        return (IonStruct) type.getIsl();
+    }
+
+    /**
+     * Get the value of constraint 'type' in IonType format.
+     * @return the value of 'type' in IonType format.
+     */
+    public IonType getIonType() {
+        return IonType.valueOf(getIsl().get(KEYWORD_TYPE).toString().toUpperCase());
+    }
```

Comment:
> This is working for now because we're only supporting basic Ion types for generation. Since later on we'll want to support [user-defined types](https://amzn.github.io/ion-schema/docs/spec.html#type-definitions) for the `type` constraint we'll ultimately have to change this- yield a string instead and then look that up by [`Schema#getType(String)`](https://www.javadoc.io/static/com.amazon.ion/ion-schema-kotlin/1.2.1/com/amazon/ionschema/Schema.html#getType(String)) so that we can recursively resolve constraints and generate.

This logic sounds better. Only one question here: Looking up type definition by [Schema#getType(String)](https://www.javadoc.io/static/com.amazon.ion/ion-schema-kotlin/1.2.1/com/amazon/ionschema/Schema.html#getType(String)), does this mean the provided type definition should be identified by name(should contain `name` constraint)? Cause the parameter required by `getType (String name) `is the `name` constraint of the type definition. And if unnamed type definition provided, we are not able to get that.